tests/ui/fmt/format_in_macro.stderr

```diff
@@ -0,0 +1,41 @@
+error: format argument must be a string literal
+  --> $DIR/format_in_macro.rs:3:10
+   |
+LL | #[derive(format_in_macro::derive)]
+   |          ^^^^^^^^^^^^^^^^^^^^^^^
+   |
+   = help: you might be missing a string literal to format with in the definition of this macro
+   = note: this error originates in the derive macro `format_in_macro::derive` (in Nightly builds, run with -Z macro-backtrace for more info)
```

Comment:
I don't see the issue from https://github.com/rust-lang/rust/issues/116190 (`#[derive("{}", macros::bare_println)]`) in the test outputs.
Could you add all the tests and make them pass in one commit, and then apply compiler fixes and corresponding test updates in the second commit?